karoo_gp/population.py

```diff
@@ -0,0 +1,2039 @@
+import functools, ast
+import operator as op
+
+import numpy as np
+import sklearn.metrics as skm
+from sympy import sympify
+import tensorflow.compat.v1 as tf
+tf.disable_v2_behavior()  # from https://www.tensorflow.org/guide/migrate
+
+from . import Tree
+
+operators = {
+    ast.Add: tf.add,  # e.g., a + b
+    ast.Sub: tf.subtract,  # e.g., a - b
+    ast.Mult: tf.multiply,  # e.g., a * b
+    ast.Div: tf.divide,  # e.g., a / b
+    ast.Pow: tf.pow,  # e.g., a ** 2
+    ast.USub: tf.negative,  # e.g., -a
+    ast.And: tf.logical_and,  # e.g., a and b
+    ast.Or: tf.logical_or,  # e.g., a or b
+    ast.Not: tf.logical_not,  # e.g., not a
+    ast.Eq: tf.equal,  # e.g., a == b
+    ast.NotEq: tf.not_equal,  # e.g., a != b
+    ast.Lt: tf.less,  # e.g., a < b
+    ast.LtE: tf.less_equal,  # e.g., a <= b
+    ast.Gt: tf.greater,  # e.g., a > b
+    ast.GtE: tf.greater_equal,  # e.g., a >= 1
+    'abs': tf.abs,  # e.g., abs(a)
+    'sign': tf.sign,  # e.g., sign(a)
+    'square': tf.square,  # e.g., square(a)
+    'sqrt': tf.sqrt,  # e.g., sqrt(a)
+    'pow': tf.pow,  # e.g., pow(a, b)
+    'log': tf.log,  # e.g., log(a)
+    'log1p': tf.log1p,  # e.g., log1p(a)
+    'cos': tf.cos,  # e.g., cos(a)
+    'sin': tf.sin,  # e.g., sin(a)
+    'tan': tf.tan,  # e.g., tan(a)
+    'acos': tf.acos,  # e.g., acos(a)
+    'asin': tf.asin,  # e.g., asin(a)
+    'atan': tf.atan,  # e.g., atan(a)
+    }
+
+# ----------------------------
+# MAY REDESIGN
+#
+# The Population and Tree objects implement basic api methods, initially as
+# wrappers for 'fx_..' functions.
+#
+# For now, they hold very little in state and take an obscene number of args.
+# This is temporary, while we figure out what should be stored where.
+
+class Population:
+
```

Comment:
Similarly, you can add a class docstring to explain what this class represent.  If the module only defines this class, in the module docstring you can be brief and just say `"""Define the Population class."""` and then expand here.
In general, modules are collections of classes and functions, so if you only have one class, maybe you don't need a separate module (also @kstaats might also want to keep everything in a single file).